src/Stratis.Bitcoin/EventBus/IEvent.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Stratis.Bitcoin.EventBus
+{
+    /// <summary>
+    /// Base event interface.
+    /// We can requires to specify a Sender if we want (who produced the event), preferably as WeakReference
+    /// </summary>
+    public interface IEvent
```

Comment:
I don't see the point of an interface just use the abstract

Reply:
it's the constraint used by generic methods, because you are not forced to use EventBase, you can have your own "base" event class

Reply:
actually I see the mistake is not constraining generic type on this interface but on EventBase, i should change that to constraint IEvent here
`SubscriptionToken Subscribe<TEventBase>(Action<TEventBase> action) where TEventBase : EventBase;`